recipes/CommonVoice/ASR/transformer/train_with_whisper.py

```diff
@@ -48,10 +48,16 @@ def compute_forward(self, batch, stage):
         enc_out, logits, _ = self.modules.whisper(wavs, bos_tokens)
 
         hyps = None
-        if stage == sb.Stage.VALID:
-            hyps, _ = self.hparams.valid_greedy_searcher(enc_out, wav_lens)
-        elif stage == sb.Stage.TEST:
-            hyps, _ = self.hparams.valid_greedy_searcher(enc_out, wav_lens)
+        if stage == sb.Stage.VALID or stage == sb.Stage.TEST:
+            # Decide searcher for inference: valid or test search
+            search = getattr(self.hparams, f"{stage.name}_search".lower())
```

Comment:
this getattrb with string parsing is AAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAH in a recipe.

Reply:
fixed